rlp/encode.go

```diff
@@ -144,6 +145,10 @@ func makeWriter(typ reflect.Type, ts rlpstruct.Tags) (writer, error) {
 		return writeBigIntPtr, nil
 	case typ.AssignableTo(bigInt):
 		return writeBigIntNoPtr, nil
+	case typ.AssignableTo(reflect.PtrTo(u256Int)):
+		return writeU256IntPtr, nil
+	case typ.AssignableTo(u256Int):
```

Comment:
We check for big.Int in the same way, but it's less of a problem because (1) big.Int is a struct with private fields, and (2) it is usually handled as a pointer, and named types can't be pointers (`type X *big.Int` is not allowed).